### ISSUE TITLE
"Extension" use in "File path" and "Image file path"

### ISSUE THREAD

**Issue by jstarpl** [SuperFlyTV/GraphicsDataDefinition#9]
"File path" and "Image file path" both refer to a `gddOptions.extensions` property, that is an array that "Limit which files can be chosen by the user". The problem I see with that is that Web APIs (and some OS as well) generally don't rely on DOS-style "file name extensions" for restricting user file selection. I feel that GDD should follow Web API standards and use an array of MIME-types and MIME-type globs for limiting file types to be selected and expect the implementation to work out how to filter on those MIME-types.

In the case of Web apps [(1)](https://developer.mozilla.org/en-US/docs/Web/API/Window/showOpenFilePicker) and Python [(2)](https://docs.python.org/3/library/mimetypes.html#mimetypes.guess_all_extensions) its free, Electron-based apps [(3)](https://www.electronjs.org/docs/latest/api/dialog#dialogshowopendialogbrowserwindow-options) would have to use a MIME-type database like [mime-types](https://www.npmjs.com/package/mime-types) if not using the Web API, same for C# [MimeTypesMap](https://www.nuget.org/packages/MimeTypesMap) - however it should be possible to interrogate OS mime-type database as well: [(4)](https://stackoverflow.com/questions/21329898/opposite-of-mimemapping-getmimemapping) and [(5)](https://developer.apple.com/documentation/uniformtypeidentifiers/uttype/3551512-init)

**Comment by nytamin** [comment]
I agree fully! I initially just added those in there as examples of what a GDD-Type could be.

We should definitely rework them

**Comment by didikunz** [comment]
While "File path" is of broader range (it can be anything) I would limit the "Image file path" to the most common image file types png, tiff, targa and maybe bmp or svg. If one needs something else he could use the standard "File path" with the corresponding mime type. That would make the implementation of a UI simpler when one only needs images.
